RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -150,12 +115,55 @@ public class PaywallViewController: UIViewController {
         super.viewDidDisappear(animated)
     }
 
+    /// - Warning: For internal use only
+    @objc(updateWithOffering:)
+    public func update(with offering: Offering) {
+        self.configuration.content = .offering(offering)
+    }
+
+    /// - Warning: For internal use only
+    @objc(updateWithOfferingIdentifier:)
+    public func update(with offeringIdentifier: String) {
+        self.configuration.content = .offeringIdentifier(offeringIdentifier)
+    }
+
+    // MARK: - Internal
+
     class var mode: PaywallViewMode {
         return .fullScreen
     }
 
+    // MARK: - Private
+
+    private var hostingController: UIHostingController<PaywallContainerView>? {
+        willSet {
+            guard let oldValue = self.hostingController else { return }
+
+            oldValue.willMove(toParent: nil)
+            oldValue.view.removeFromSuperview()
+            oldValue.removeFromParent()
+        }
+
+        didSet {
+            guard let newController = self.hostingController else { return }
```

Comment:
given that this is not the most simple or intuitive implementation, we should document _why_ we're doing this

Reply:
👍🏻